src/riak_kv_qry_compiler.erl

```diff
@@ -4195,4 +4262,263 @@ query_with_desc_last_local_key_column_no_quantum_test() ->
+    {ok, SubQueries} = compile(DDL, Q),
+    ?assertEqual(
+        [[{startkey,[{<<"a">>,timestamp,10}]},
+          {endkey,  [{<<"a">>,timestamp,10}]},
+          {filter, {'=',{field,<<"b">>,double},{const,6.0}}},
+          {end_inclusive, true}]],
+        [W || ?SQL_SELECT{'WHERE' = W} <- SubQueries]
+    ).
+
+query_with_arithmetic_in_where_clause_operator_precedence_queries_equal_test() ->
+    DDL = get_ddl(
+        "CREATE table table1 ("
+        "a VARCHAR NOT NULL,"
+        "b TIMESTAMP NOT NULL,"
+        "PRIMARY KEY ((a, b), a, b));"
+    ),
+    {ok, QA} = get_query(
+          "SELECT * FROM table1 WHERE a = 'hi' AND b = 2+3*4"),
+    {ok, QB} = get_query(
+          "SELECT * FROM table1 WHERE a = 'hi' AND b = 3*4+2"),
+    {ok, SubQueriesA} = compile(DDL, QA),
+    {ok, SubQueriesB} = compile(DDL, QB),
+    ?assertEqual(
+        [W || ?SQL_SELECT{'WHERE' = W} <- SubQueriesA],
+        [W || ?SQL_SELECT{'WHERE' = W} <- SubQueriesB]
+    ).
+
+select_with_arithmetic_on_identifier_throws_an_error_test() ->
+    DDL = get_ddl(
+        "CREATE table table1 ("
+        "a SINT64 NOT NULL,"
+        "PRIMARY KEY ((a), a));"
+    ),
+    {ok, Q} = get_query(
+        "SELECT * FROM table1 "
+        "WHERE a+1 = 10"),
+    ?assertEqual(
+        {false,[{arithmetic_on_identifier,<<"a">>}]},
+        is_query_valid(DDL, Q)
+    ).
+
+select_with_arithmetic_on_identifier_throws_an_error_literal_on_lhs_test() ->
+    DDL = get_ddl(
+        "CREATE table table1 ("
+        "a SINT64 NOT NULL,"
+        "PRIMARY KEY ((a), a));"
+    ),
+    {ok, Q} = get_query(
+        "SELECT * FROM table1 "
+        "WHERE 10 = a+1"),
+    ?assertEqual(
+        {false,[{arithmetic_on_identifier,<<"a">>}]},
+        is_query_valid(DDL, Q)
+    ).
+
+%%
+%% Test macro for assertions on now and arithmetic, shouldn't be
+%% extended for other uses
+%%
+-define(assertQueryCompilesToFilter(Options,QuerySQL,Filter),
+    DDL = get_ddl(
+        "CREATE table table1 ("
+        "a VARCHAR NOT NULL,"
+        "b TIMESTAMP NOT NULL,"
+        "c TIMESTAMP NOT NULL,"
+        "PRIMARY KEY ((a, b), a, b));"
+    ),
+    {ok, Q} = get_query(QuerySQL),
+    {ok, SubQueries} = compile(DDL, Q, Options),
+    ?assertEqual(
+        [[{startkey,[{<<"a">>,varchar,<<"hi">>},{<<"b">>,timestamp,4000}]},
+          {endkey,  [{<<"a">>,varchar,<<"hi">>},{<<"b">>,timestamp,4000}]},
+          {filter,  Filter},
+          {end_inclusive, true}]],
+        [W || ?SQL_SELECT{'WHERE' = W} <- SubQueries]
+    )).
+now_function_in_where_test() ->
+    ?assertQueryCompilesToFilter(
+        [{now_milliseconds,1979}],
+        "SELECT * FROM table1 WHERE a = 'hi' AND b = 4000 AND c > now()",
+        {'>',{field,<<"c">>,timestamp},{const,1979}}
+    ).
+now_function_in_where_with_addition_test() ->
+    ?assertQueryCompilesToFilter(
+        [{now_milliseconds,1979}],
+        "SELECT * FROM table1 WHERE a = 'hi' AND b = 4000 AND c > now()+1",
```

Comment:
`"SELECT * FROM table1 WHERE a = 'hi' AND b = 4000 AND c > now()+1*2"`. I don't contend there _is_ an issue; I just suggest you have it covered in the unit tests.